### DISCUSSION TITLE
What's your favorite workflow?

### DISCUSSION THREAD

**Discussion by X-Y** [tailwindlabs/tailwindcss#8638] [Help] [answered]
Hey guys! I'm new to Tailwind. Usually when I'm working on CSS, I would inspect the element in browser Devtools, edit the styling, copy them to the css files, then do some clean-ups. With TailwindCSS, I'd have to translate the mocked css to tailwind rules, and add them to the classnames. Working with react, the browser auto-reloads whenever there's a change in the source file, so it gets a bit annoying. I'm wondering if you have a better workflow?

**Comment by wanacode** [comment, marked as answer]
Rather than writing the css in Devtools, with Tailwind you are writing the css directly in the html as utility classes. 

So if you are working on a button you might add some vertical and horizontal padding, `px-12 py-3`.

Add some foreground and background color, `text-white bg-indigo-600`.

Even add some hover state, `hover:bg-white hover:text-indigo-600`.

If the browser is reloading that is ideal as you can see how it looks as you add the utility classes.

And a few more classes, even for different break points, and BOOM, you have a button, https://play.tailwindcss.com/OF5qKE1EXb

No copy and pasting css into a separate stylesheet. Just on to the next component.

**Reply by X-Y** [reply]
My issue is that, as a new user to TailwindCSS, trying to look up all the rules first would slow me down a lot, comparing to write them in vanilla CSS first. 

It's also much faster to test out styling in inspection panel. I can add/remove rules, change numeric values with arrow keys, all in the same window and see the changes in real time. Other than changing them in IDE, wait for it to update, inspect element to see what is wrong, and go back to IDE to make another change.

**Reply by wanacode** [reply]
Sure you have to learn the 'rules' but I found I was productive after only a day or two. Two things that helped me were the VS Code Tailwind CSS IntelliSense (https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) and also this cheatsheet (although it hasn't been updated for V3) https://nerdcave.com/tailwind-cheat-sheet

I am generally converting existing designs to code, so if you are designing you will have a different workflow. I would think you would still be more productive with Tailwind, especially when you take into account writing code for breakpoints and hover state etc. 

Maybe it is not for you, but I would at least give it a chance. :)
